sendgrid.go

```diff
@@ -48,6 +48,82 @@ func NewSendClient(key string) *Client {
 	return &Client{request}
 }
 
+// NewClientForEndpoint returns a client that can send requests to a specific endpoint.
+func NewClientForEndpoint(key, endpoint string) *Client {
+	request := GetRequest(key, endpoint, "")
+	return &Client{request}
+}
+
+// DefaultClient is used if no custom HTTP client is defined
```

Comment:
I think you should be able to remove the following functions, as they now reside here: https://github.com/sendgrid/sendgrid-go/blob/main/base_interface.go